CMakeLists.txt

```diff
@@ -205,6 +207,92 @@ if (NN_ENABLE_NANOCAT)
     target_link_libraries (nanocat ${PROJECT_NAME})
 endif ()
 
+# Build the documenation
+if (NN_ENABLE_DOC)
+    find_program (ASCIIDOCTOR_EXE asciidoctor)
+    if (NOT ASCIIDOCTOR_EXE)
+        message (SEND_ERROR "asciidoctor not found, cannot create docs")
+    endif ()
```

Comment:
According to CMake docs,

CMake Error, continue processing,
                              but skip generation

So I think this means that further operation inside cmake will continue,
but it won’t actually emit a Makefile (or whatever).  This way we can check
for other errors too.  Am I mistaken?


On Fri, Apr 22, 2016 at 3:56 PM, Franklin Mathieu <notifications@github.com>
wrote:

> In CMakeLists.txt
> <https://github.com/nanomsg/nanomsg/pull/665#discussion_r60813444>:
>
> > @@ -205,6 +207,92 @@ if (NN_ENABLE_NANOCAT)
> >      target_link_libraries (nanocat ${PROJECT_NAME})
> >  endif ()
> >
> > +# Build the documenation
> > +if (NN_ENABLE_DOC)
> > +    find_program (ASCIIDOCTOR_EXE asciidoctor)
> > +    if (NOT ASCIIDOCTOR_EXE)
> > +        message (SEND_ERROR "asciidoctor not found, cannot create docs")
> > +    endif ()
>
> SEND_ERROR continues execution, so the whole block below should be
> skipped if asciidoctor isn't found.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nanomsg/nanomsg/pull/665/files/55921246a02a8d9581fbb6e961a6aae3aadbb2c7#r60813444>
>